### PR TITLE
Remove deprecated `.extend` transformation pattern

### PR DESCRIPTION
**Pull request by renatonouman**
This PR makes styled-components `.extend` pattern optional, since it has been deprecated in v4(styled-components/styled-components#1908).

Moreover, appending `.withConfig` to other extend methods might break unrelated code, i.e. https://github.com/Igorbek/typescript-plugin-styled-components/issues/33

Related to #47, #33 and #48 

From #48 
- [x] make extend identifier configurable like other identifiers;
- [x] set default value to an empty array;
- [x] add readme/release notes with breaking change information and instructions to re-enable by providing settings.

### CODE REVIEW

**Comment by renatonouman** [comment]
@Igorbek any instructions on how to add this to readme and release notes?

File: src/createTransformer.ts [hunk 1]
```diff
@@ -28,9 +28,7 @@ function isStyledFunction(node: ts.Node, identifiers: CustomStyledIdentifiers):
             return true;
         }
 
-        if (node.name.text === 'extend'
-            && isValidComponent(node.expression)) {
-
+        if (isStyledExtendIdentifier(node.name.text, identifiers)) {
```
Comment (Igorbek) [inline src/createTransformer.ts:31]:
```suggestion
        if (isStyledExtendIdentifier(node.name.text, identifiers)
            && isValidComponent(node.expression)) {
```

we still need to check that it is a valid component.

**Review by Igorbek — CHANGES_REQUESTED** [review]
Thank you for the contribution! Could you please add a line to the 'unreleased' section to the `CHANGELOG.md` file. I just created it.

**Comment by renatonouman** [comment]
Thank you for reviewing this so quickly. Changes made as requested!

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,8 @@ All notable changes to this project will be documented in this file.
 
 ## [Unreleased]
 
+- remove deprecated .extend transformation pattern (fix #48 in #152)
```
Comment (Igorbek) [inline CHANGELOG.md:6]:
```suggestion
- remove deprecated .extend transformation pattern (fix #48 in #370)
```

**Review by Igorbek — CHANGES_REQUESTED** [review]

**Review by Igorbek — APPROVED** [review]

**Comment by Igorbek** [comment]
Sorry, @renatonouman another pr got merged, please resolve conflicts and update snapshots.

File: CHANGELOG.md (hunk 2 above)
Comment (renatonouman) [reply CHANGELOG.md]: Oh, sorry about this. Fixed!

**Comment by renatonouman** [comment]
Done @Igorbek

**Review by Igorbek — APPROVED** [review]

**Comment by renatonouman** [comment]
Glad to contribute! Thanks for the review and merge!

**Comment by renatonouman** [comment]
Hey @Igorbek! I just want to check with you when do you plan on releasing the next version?
Thanks!

**Comment by Igorbek** [comment]
I started preparing it lately, but got interrupted with other stuff, as usual. I'll try to release it tonight. Sorry for inconvenience.

**Comment by Igorbek** [comment]
It's now been released https://github.com/Igorbek/typescript-plugin-styled-components/releases/tag/1.5.0
Thank you for the contribution @renatonouman!

**Comment by renatonouman** [comment]
Thank you for the release @Igorbek.
Using the new version in my project already!

**Pull request merged**